microproxy/config.py

```diff
@@ -167,18 +179,17 @@ def type_transform(config, optionInfo):
     for field in config:
         if field not in optionInfo:
             continue
+        option_type = optionInfo[field]["type"].split(":")[0]
+        type_func = getattr(__builtin__, option_type)
 
-        if optionInfo[field]["type"] == "int":
-            new_config[field] = int(config[field])
-
-        elif optionInfo[field]["type"] == "list":
+        if "list" == option_type:
+            list_type = optionInfo[field]["type"].split(":")[1]
+            type_func = getattr(__builtin__, list_type)
```

Comment:
I think it could just define a list of support type, ex: list, str, int, bool etc.
And we only handle for the fix set of type.
I think there is no need for the flexibility that sacrifice some of the readability.